.github/workflows/check-mergability.yaml

```diff
@@ -0,0 +1,38 @@
+# Checks for merge conflicts when a pull request is opened or updated
+name: Check mergability onto main
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  merge_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
```

Comment:
Ok, sorry, looks like we indeed need full history here for merge to work correctly.